apps/meteor/client/views/marketplace/AppsPage/AppsPageContent.tsx

```diff
@@ -186,6 +188,30 @@ const AppsPageContent = (): ReactElement => {
 		toggleInitialSortOption(isRequested);
 	}, [isMarketplace, isRequested, sortFilterOnSelected, t, toggleInitialSortOption]);
 
+	const getEmptyState = () => {
```

Comment:
I think is better to encapsulate this function in a `React.useMemo` here...

Reply:
Not sure I agree, whenever the component is rerendered this function should be rerendered as well. From my point of view the operations in the function are lightweight enough (all are synchronous, simple if statements based on existing data) so that it does not justify the use of `useMemo`

Reply:
Every single top change that doesn't affect the state of the return of `getEmptyState` could re-trigger another re-computation of values unnecessary... Are you sure we'll not have any scenario like this?

Reply:
I am quite confident that the performance of encapsulating and retrieving the value of this function in a `useMemo` and simply executing the function again on every rerender will be very similar, with negligible difference. Also, memoizing the value will use more memory as React will need to store the value with the dependencies as keys for it.

If it was the case that this function would iterate over a large set of data, or require data from elsewhere asynchronously, I would say `useMemo` could help. In this specific case, I don't see benefits on using it